src/routes/api/playlists/user/+server.ts

```diff
@@ -0,0 +1,14 @@
+import { authenticate, getUserData, readPath } from '$lib/server/firebaseUtils';
+import { json } from '@sveltejs/kit';
+import type { RequestHandler } from './$types';
+
+export const GET = (async (event) => {
+    const { uid } = await authenticate(event);
+    const playlists = await readPath(`/playlists/user/${uid}`) || {};
+    const playlistArray = Object.entries(playlists).map(([key, val]) => val);
+    return json(playlistArray, { headers: [ ['Access-Control-Allow-Origin', "*"] ]});
+}) satisfies RequestHandler;
+
+export const OPTIONS = (() => {
+    return new Response(null, { headers: [[ 'Access-Control-Allow-Origin', "*" ], [ 'Access-Control-Allow-Headers', '*' ]]});
```

Comment:
I'm not sure that I have always done this, but I think we should explicitly allow only the methods available, so in this case
```suggestion
    return new Response(null, { headers: [[ 'Access-Control-Allow-Origin', "*" ], [ 'Access-Control-Allow-Headers', '*' ], ['Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE']]});
```

Reply:
roger, making this change!